#### 1. **Importación de librerías y carga de datos**

In [ ]:
# Importación de librerías necesarias.
import pandas as pd
import numpy as np
import os
import sys 
sys.path.append('../')
from src import sp_analisis as sa
from src import sp_rutas as ru  # Archivo de soporte para rutas: sp_rutas.py -> Cambiar ruta "ruta.raw_data" según sitema operativo.

# Configuración para mostrar todas las columnas.
pd.set_option('display.max_columns', None)

#### 2. **Abrimos los archivos uno a uno y hacemos un primer preanálisis para obtener las dimensiones y la información más relevante de nuestro archivos**

In [2]:
ruta = ru.ruta_raw_data+"/categories.csv" # Ruta del archivo CSV. Cambiar según sea necesario.
df_categories = pd.read_csv(ruta)
sa.preanalisis(df_categories)

El dataframe tiene 11 filas y 2 columnas
--------------------------------------------------
Tenemos un total de 0 valores duplicados
--------------------------------------------------
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11 entries, 0 to 10
Data columns (total 2 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   CategoryID    11 non-null     int64 
 1   CategoryName  11 non-null     object
dtypes: int64(1), object(1)
memory usage: 308.0+ bytes


None

--------------------------------------------------
No hay valores nulos
--------------------------------------------------
Los estadísticos descriptivos de las variables numéricas son:


count  mean       std  min  25%  50%  75%   max
CategoryID   11.0   6.0  3.316625  1.0  3.5  6.0  8.5  11.0

--------------------------------------------------
Los estadísticos descriptivos de las variables categóricas son:


count unique          top freq
CategoryName    11     11  Confections    1

--------------------------------------------------
La distribución de frecuencias de las variables categóricas es:

Para la columna CATEGORYNAME, los valores únicos son:
['Confections' 'Shell fish' 'Cereals' 'Dairy' 'Beverages' 'Seafood' 'Meat'
 'Grain' 'Poultry' 'Snails' 'Produce']


Frecuencia  Porcentaje
CategoryName                        
Confections            1        9.09
Shell fish             1        9.09
Cereals                1        9.09
Dairy                  1        9.09
Beverages              1        9.09
Seafood                1        9.09
Meat                   1        9.09
Grain                  1        9.09
Poultry                1        9.09
Snails                 1        9.09
Produce                1        9.09

--------------------------------------------------


Análisis inicial del archivo "categories":

1. Tipo de datos a cambiar: no es necesario hacer ningún cambio

2. Categorías a cambiar: 

    - Cambiar el nombre de las columnas:

        - **CategoryId** -> Category_Id
        - **CategoryName** -> Category_Name

3. Columnas con valores nulos: no hay

4. Columnas a eliminar: no es necesario eliminar ninguna columna

In [3]:
ruta = ru.ruta_raw_data+"/cities.csv" # Ruta del archivo CSV. Cambiar según sea necesario.
df_cities = pd.read_csv(ruta)
sa.preanalisis(df_cities)

El dataframe tiene 96 filas y 4 columnas
--------------------------------------------------
Tenemos un total de 0 valores duplicados
--------------------------------------------------
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 96 entries, 0 to 95
Data columns (total 4 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   CityID     96 non-null     int64 
 1   CityName   96 non-null     object
 2   Zipcode    96 non-null     int64 
 3   CountryID  96 non-null     int64 
dtypes: int64(3), object(1)
memory usage: 3.1+ KB


None

--------------------------------------------------
No hay valores nulos
--------------------------------------------------
Los estadísticos descriptivos de las variables numéricas son:


count          mean           std    min       25%      50%  \
CityID      96.0     48.500000     27.856777    1.0     24.75     48.5   
Zipcode     96.0  51245.302083  29883.397049  157.0  21780.50  51700.0   
CountryID   96.0     32.000000      0.000000   32.0     32.00     32.0   

                75%      max  
CityID        72.25     96.0  
Zipcode    79867.75  97859.0  
CountryID     32.00     32.0

--------------------------------------------------
Los estadísticos descriptivos de las variables categóricas son:


count unique     top freq
CityName    96     96  Dayton    1

--------------------------------------------------
La distribución de frecuencias de las variables categóricas es:

Para la columna CITYNAME, los valores únicos son:
['Dayton' 'Buffalo' 'Chicago' 'Fremont' 'Virginia Beach' 'Austin'
 'Riverside' 'Arlington' 'Atlanta' 'Toledo' 'Spokane' 'Tacoma' 'Akron'
 'Indianapolis' 'Colorado' 'Norfolk' 'St. Louis' 'Little Rock' 'Omaha'
 'El Paso' 'Milwaukee' 'Baton Rouge' 'Anaheim' 'Lincoln' 'Boston'
 'Des Moines' 'Mesa' 'Tucson' 'Washington' 'Philadelphia' 'Wichita'
 'Jackson' 'Minneapolis' 'Fort Wayne' 'Greensboro' 'Detroit' 'Fresno'
 'Richmond' 'Lubbock' 'Cleveland' 'Bakersfield' 'Denver' 'Glendale'
 'Memphis' 'Dallas' 'New York' 'San Francisco' 'Long Beach' 'Montgomery'
 'Cincinnati' 'Portland' 'Madison' 'Las Vegas' 'Albuquerque' 'Houston'
 'Stockton' 'Nashville' 'Columbus' 'Kansas' 'Fort Worth' 'Miami' 'Mobile'
 'Tulsa' 'Raleigh' 'Baltimore' 'Shreveport' 'Corpus Christi' 'Anchorage'
 'Oakland' 'Rochester' 'Phoenix' 'Honolulu' 'Santa Ana' 'Tampa'

Frecuencia  Porcentaje
CityName                              
Dayton                   1        1.04
Buffalo                  1        1.04
Chicago                  1        1.04
Fremont                  1        1.04
Virginia Beach           1        1.04
...                    ...         ...
Hialeah                  1        1.04
Jersey                   1        1.04
Yonkers                  1        1.04
Garland                  1        1.04
Pittsburgh               1        1.04

[96 rows x 2 columns]

--------------------------------------------------


Análisis inicial del archivo "cities":

1. Tipo de datos a cambiar: no es necesario hacer ningún cambio

2. Categorías a cambiar: 

    - Cambiar nombre de las columnas:
    
        - **CityID** -> City_Id
        - **CityName** -> City_Name
        - **CountryID** -> Country_Id

3. Columnas con valores nulos: no hay

4. Columnas a eliminar: no es necesario eliminar ninguna columna

In [4]:
ruta = ru.ruta_raw_data+"/countries.csv" # Ruta del archivo CSV. Cambiar según sea necesario.
df_countries = pd.read_csv(ruta)
sa.preanalisis(df_countries)

El dataframe tiene 206 filas y 3 columnas
--------------------------------------------------
Tenemos un total de 0 valores duplicados
--------------------------------------------------
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 206 entries, 0 to 205
Data columns (total 3 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   CountryID    206 non-null    int64 
 1   CountryName  206 non-null    object
 2   CountryCode  205 non-null    object
dtypes: int64(1), object(2)
memory usage: 5.0+ KB


None

--------------------------------------------------
Los nulos por columna se distribuyen de la siguiente forma:


Valores_nulos  Porcentaje_nulos
CountryCode              1              0.49

--------------------------------------------------
Los estadísticos descriptivos de las variables numéricas son:


count   mean        std  min    25%    50%     75%    max
CountryID  206.0  103.5  59.611241  1.0  52.25  103.5  154.75  206.0

--------------------------------------------------
Los estadísticos descriptivos de las variables categóricas son:


count unique      top freq
CountryName   206    206  Armenia    1
CountryCode   205    205       AN    1

--------------------------------------------------
La distribución de frecuencias de las variables categóricas es:

Para la columna COUNTRYNAME, los valores únicos son:
['Armenia' 'Canada' 'Belize' 'Uganda' 'Thailand' 'Tunisia' 'Montserrat'
 'Iraq' 'Slovakia' 'Germany' 'Mauritania' 'Israel' 'Panama' 'Kazakhstan'
 'Swaziland' 'Malawi' 'Malta' 'Czech Republic' 'Mauritius' 'Luxembourg'
 'Macedonia' 'Haiti' 'Saint Helena' 'Austria' 'Bahrain' 'Denmark' 'Zambia'
 'Paraguay' 'Oman' 'Honduras' 'Congo' 'United States' 'Comoros' 'Macao'
 'Uruguay' 'Moldova' 'Vatican City' 'Liechtenstein' 'Guatemala' 'Bermuda'
 'Sudan' 'Guinea' 'Cape Verde' 'Liberia' 'Libya' 'Tuvalu' 'Kenya'
 'Finland' 'Azerbaijan' 'Aruba' 'San Marino' 'Italy' 'Djibouti'
 'Isle of Man' 'Dominica' 'Virgin Islands' 'Benin' 'South Georgia'
 'Grenada' 'Philippines' 'Cook Islands' 'Guyana' 'Lesotho' 'Somalia'
 'Lebanon' 'Greece' 'Samoa' 'Estonia' 'Argentina' 'Namibia' 'Suriname'
 'Algeria' 'Portugal' 'Serbia' 'France' 'Tonga' 'Jamaica

Frecuencia  Porcentaje
CountryName                        
Armenia               1        0.49
Canada                1        0.49
Belize                1        0.49
Uganda                1        0.49
Thailand              1        0.49
...                 ...         ...
Greenland             1        0.49
Niger                 1        0.49
Malvinas              1        0.49
Mayotte               1        0.49
Mongolia              1        0.49

[206 rows x 2 columns]

--------------------------------------------------
Para la columna COUNTRYCODE, los valores únicos son:
['AN' 'FO' 'MK' 'LV' 'VI' 'GF' 'MM' 'AO' 'CG' 'TT' 'PM' 'SV' 'BE' 'MS'
 'MP' 'MV' 'BO' 'AF' 'CZ' 'LB' 'MA' 'SG' 'IN' 'CM' 'YT' 'TN' 'LK' 'BB'
 'DZ' 'JO' 'TR' 'AR' 'LR' 'RE' 'IO' 'GL' 'NF' 'LU' 'ES' 'AG' 'FJ' 'PW'
 'AQ' 'SZ' 'DJ' 'EC' 'BV' 'CV' 'GU' 'AU' 'AS' 'AZ' 'SR' 'KG' 'IL' 'KY'
 'TF' 'VU' 'NZ' 'RO' 'DM' 'MT' 'VG' 'FR' 'LA' 'TJ' 'GN' 'NG' 'CO' 'TC'
 'CK' 'NE' 'VE' 'KP' 'ZM' 'HU' 'JP' 'ML' 'GQ' 'TD' 'FI' 'PH' 'BJ' 'KR'
 'NO' 'BM' 'YU' 'MG' 'ST' 'HT' 'AM' 'ZA' 'BZ' 'SO' 'MD' 'CC' 'BG' 'JM'
 'CY' 'PR' 'EH' 'NI' 'TK' 'CF' 'MO' 'SE' 'KN' 'GB' 'BH' 'BT' 'AW' 'PN'
 'LI' 'HN' 'UY' 'BS' 'QA' 'KH' 'BI' 'SK' 'HM' 'EE' 'SL' 'GM' 'TH' 'GA'
 'IE' 'SM' 'CR' 'LS' 'MX' 'KM' 'SD' 'SI' 'AD' 'RW' 'MQ' 'DK' 'US' 'IT'
 'GP' 'CI' 'WS' 'GY' 'TG' nan 'ZW' 'BF' 'TV' 'CA' 'ID' 'BW' 'TO' 'HK' 'KZ'
 'CN' 'CH' 'FK' 'CX' 'AT' 'GT' 'GW' 'PT' 'GR' 'LT' 'PF' 'VN' 'AE' 'TW'
 'KE' 'CL' 'NL' 'MC' 'KI' 'SC' 'BR' 'OM'

Frecuencia  Porcentaje
CountryCode                        
AN                    1        0.49
FO                    1        0.49
MK                    1        0.49
LV                    1        0.49
VI                    1        0.49
...                 ...         ...
MW                    1        0.49
SH                    1        0.49
GE                    1        0.49
PG                    1        0.49
MZ                    1        0.49

[205 rows x 2 columns]

--------------------------------------------------


Análisis inicial del archivo "countries":

1. Tipo de datos a cambiar: no es necesario hacer ningún cambio

2. Categorías a cambiar: 

    - Cambiar nombre de las columnas:

        - **CountryID** -> Country_Id
        - **CountryName** -> Country_Name
        - **CountryCode** -> Country_Code

3. Columnas con valores nulos: 

    - **CountryCode**: 1 valor nulo

4. Columnas a eliminar: **CountryCode** no aporta valor a nuestro análisis

In [5]:
ruta = ru.ruta_raw_data+"/customers.csv" # Ruta del archivo CSV. Cambiar según sea necesario.
df_customers = pd.read_csv(ruta)
sa.preanalisis(df_customers)

El dataframe tiene 98759 filas y 6 columnas
--------------------------------------------------
Tenemos un total de 0 valores duplicados
--------------------------------------------------
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 98759 entries, 0 to 98758
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   CustomerID     98759 non-null  int64 
 1   FirstName      98759 non-null  object
 2   MiddleInitial  97782 non-null  object
 3   LastName       98759 non-null  object
 4   CityID         98759 non-null  int64 
 5   Address        98759 non-null  object
dtypes: int64(2), object(4)
memory usage: 4.5+ MB


None

--------------------------------------------------
Los nulos por columna se distribuyen de la siguiente forma:


Valores_nulos  Porcentaje_nulos
MiddleInitial            977              0.99

--------------------------------------------------
Los estadísticos descriptivos de las variables numéricas son:


count          mean           std  min      25%      50%  \
CustomerID  98759.0  49380.000000  28509.411955  1.0  24690.5  49380.0   
CityID      98759.0     48.557661     27.660082  1.0     25.0     49.0   

                75%      max  
CustomerID  74069.5  98759.0  
CityID         72.0     96.0

--------------------------------------------------
Los estadísticos descriptivos de las variables categóricas son:


count unique             top  freq
FirstName      98759    969           Randi   141
MiddleInitial  97782     26               J  3826
LastName       98759    995          Dorsey   134
Address        98759  80134  16 Milton Road    10

--------------------------------------------------
La distribución de frecuencias de las variables categóricas es:

Para la columna FIRSTNAME, los valores únicos son:
['Stefanie' 'Sandy' 'Lee' 'Regina' 'Daniel' 'Dennis' 'Myra' 'Teddy'
 'Annie' 'Herman' 'Jenifer' 'Chastity' 'Evelyn' 'Jeannie' 'Rickey'
 'Bobbie' 'Lesley' 'Shawna' 'Theresa' 'Tasha' 'Karla' 'Otis' 'Alisa'
 'Peggy' 'Lisa' 'Vicky' 'Nicolas' 'Miranda' 'Kara' 'Rosemary' 'Ernest'
 'Lindsay' 'Lorenzo' 'Tammie' 'Kareem' 'Cesar' 'Charlene' 'Sonja'
 'Gwendolyn' 'Stephan' 'Maggie' 'Earnest' 'Zachary' 'Erin' 'Rodolfo'
 'Carla' 'Norma' 'Jean' 'Tara' 'Kirsten' 'Marie' 'Devin' 'Shaun' 'Jose'
 'Meredith' 'Heather' 'Sherri' 'Theodore' 'Elton' 'Phillip' 'Raymond'
 'Alejandro' 'Kristina' 'Chandra' 'Allen' 'Myron' 'Lewis' 'Laura'
 'Wallace' 'Johnathan' 'Dianna' 'Jamal' 'Sharon' 'Clifton' 'Sean' 'Trent'
 'Victor' 'Becky' 'Lana' 'Summer' 'Corey' 'Olivia' 'Miriam' 'Rosa' 'Marci'
 'Morgan' 'Gerard' 'Paul' 'Shane' 'Stanley' 'Brendan' 'Tania' 'Abr

Frecuencia  Porcentaje
FirstName                        
Randi             141        0.14
Joan              137        0.14
Michele           130        0.13
Kari              129        0.13
Douglas           128        0.13
...               ...         ...
Jeffrey            78        0.08
Melody             78        0.08
Jo                 75        0.08
Forrest            74        0.07
Liza               71        0.07

[969 rows x 2 columns]

--------------------------------------------------
Para la columna MIDDLEINITIAL, los valores únicos son:
['Y' 'T' 'S' 'H' 'K' 'L' 'F' 'R' 'N' 'Z' 'B' 'E' 'D' 'W' 'I' 'C' 'G' nan
 'Q' 'M' 'V' 'P' 'O' 'X' 'A' 'U' 'J']


Frecuencia  Porcentaje
MiddleInitial                        
J                    3826        3.87
M                    3823        3.87
R                    3812        3.86
Z                    3809        3.86
T                    3806        3.85
Y                    3804        3.85
B                    3785        3.83
X                    3778        3.83
K                    3774        3.82
C                    3773        3.82
F                    3767        3.81
P                    3766        3.81
A                    3764        3.81
Q                    3763        3.81
W                    3761        3.81
S                    3759        3.81
D                    3747        3.79
N                    3746        3.79
H                    3745        3.79
O                    3729        3.78
E                    3721        3.77
V                    3715        3.76
G                    3706        3.75
U                    3705        3.75
I                    3701        3.75
L                    3697        3.74

--------------------------------------------------
Para la columna LASTNAME, los valores únicos son:
['Frye' 'Kirby' 'Zhang' 'Avery' 'Mccann' 'Zuniga' 'Jefferson' 'Parsons'
 'Stein' 'Pope' 'Winters' 'Gordon' 'Gillespie' 'Sweeney' 'Gardner' 'Ho'
 'Simon' 'Orr' 'Herring' 'Berg' 'Heath' 'Sutton' 'Miller' 'Ruiz'
 'Sandoval' 'Dennis' 'Russell' 'Reese' 'Armstrong' 'Beltran' 'Manning'
 'Wright' 'Santana' 'Tucker' 'Gaines' 'Gonzalez' 'Smith' 'Hartman'
 'Griffith' 'Love' 'Alexander' 'Cross' 'Greer' 'Odom' 'Ward' 'Gray'
 'Acosta' 'Gallegos' 'Lozano' 'Hernandez' 'Hamilton' 'Todd' 'Gamble'
 'Singh' 'Summers' 'Larsen' 'Meza' 'Cardenas' 'Spencer' 'Harvey' 'Hurley'
 'Price' 'Barron' 'Hammond' 'Hensley' 'Best' 'Robertson' 'Dickson' 'Park'
 'Mc Cormick' 'Montoya' 'Morrison' 'Bradshaw' 'Stewart' 'Murphy' 'Lane'
 'Davis' 'Dyer' 'Mc Knight' 'Cooke' 'Sloan' 'Elliott' 'Hampton' 'Fox'
 'Cantrell' 'Dodson' 'Rich' 'Bray' 'Castillo' 'Herman' 'Chung' 'Doyle'
 'Kane' 'Mccall' 'Huang' 'Randall' 'Lynch' 'Grant' 'Be

Frecuencia  Porcentaje
LastName                         
Dorsey            134        0.14
Fisher            133        0.13
Snow              130        0.13
Dickson           127        0.13
Fritz             127        0.13
...               ...         ...
Gutierrez          76        0.08
Adkins             75        0.08
Malone             73        0.07
Leonard            73        0.07
Madden             71        0.07

[995 rows x 2 columns]

--------------------------------------------------
Para la columna ADDRESS, los valores únicos son:
['97 Oak Avenue' '52 White First Freeway' '921 White Fabien Avenue' ...
 '414 White Second Blvd.' '945 Oak Parkway' '99 Fabien Street']


Frecuencia  Porcentaje
Address                                             
16 Milton Road                        10        0.01
97 Clarendon Drive                     9        0.01
75 Oak Way                             9        0.01
39 Second Drive                        9        0.01
58 Milton St.                          8        0.01
...                                  ...         ...
442 Rocky New Freeway                  1        0.00
64 Milton Road                         1        0.00
13 Green Cowley Parkway                1        0.00
801 East White Oak Boulevard           1        0.00
30 White Oak Way                       1        0.00

[80134 rows x 2 columns]

--------------------------------------------------


Análisis inicial del archivo "customers":

1. Tipo de datos a cambiar: no es necesario hacer ningún cambio

2. Categorías a cambiar:

    - Cambiar nombre de las columnas:

        - **CustomerID** -> Customer_Id
        - **CityID** -> City_Id

3. Columnas con valores nulos: 

    - **MiddleInitial**: 977 valores nulos

4. Columnas a eliminar: 
    - Eliminar 'MiddleInitial' ya que no es necesaria para nuestro análisis
    - Concatenar las columnas 'FirstName' y 'LastName' en una sola, 'Full_Name' y eliminar las otras 2
    - Eliminar 'Address' ya que no es necesaria para nuestro análisis

In [6]:
ruta = ru.ruta_raw_data+"/employees.csv" # Ruta del archivo CSV. Cambiar según sea necesario.
df_employees = pd.read_csv(ruta)
sa.preanalisis(df_employees)

El dataframe tiene 23 filas y 8 columnas
--------------------------------------------------
Tenemos un total de 0 valores duplicados
--------------------------------------------------
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23 entries, 0 to 22
Data columns (total 8 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   EmployeeID     23 non-null     int64 
 1   FirstName      23 non-null     object
 2   MiddleInitial  23 non-null     object
 3   LastName       23 non-null     object
 4   BirthDate      23 non-null     object
 5   Gender         23 non-null     object
 6   CityID         23 non-null     int64 
 7   HireDate       23 non-null     object
dtypes: int64(2), object(6)
memory usage: 1.6+ KB


None

--------------------------------------------------
No hay valores nulos
--------------------------------------------------
Los estadísticos descriptivos de las variables numéricas son:


count       mean        std  min   25%   50%   75%   max
EmployeeID   23.0  12.000000   6.782330  1.0   6.5  12.0  17.5  23.0
CityID       23.0  43.782609  26.381737  4.0  20.5  39.0  65.0  92.0

--------------------------------------------------
Los estadísticos descriptivos de las variables categóricas son:


count unique                      top freq
FirstName        23     22                 Chadwick    2
MiddleInitial    23     14                        E    3
LastName         23     23                   Fuller    1
BirthDate        23     23  1981-03-07 00:00:00.000    1
Gender           23      2                        M   15
HireDate         23     23  2011-06-20 07:15:36.920    1

--------------------------------------------------
La distribución de frecuencias de las variables categóricas es:

Para la columna FIRSTNAME, los valores únicos son:
['Nicole' 'Christine' 'Pablo' 'Darnell' 'Desiree' 'Holly' 'Chadwick'
 'Julie' 'Daphne' 'Jean' 'Sonya' 'Lindsay' 'Katina' 'Wendi' 'Kari' 'Seth'
 'Warren' 'Bernard' 'Shelby' 'Devon' 'Tonia' 'Janet']


Frecuencia  Porcentaje
FirstName                        
Chadwick            2        8.70
Nicole              1        4.35
Pablo               1        4.35
Darnell             1        4.35
Desiree             1        4.35
Christine           1        4.35
Holly               1        4.35
Julie               1        4.35
Daphne              1        4.35
Jean                1        4.35
Sonya               1        4.35
Lindsay             1        4.35
Katina              1        4.35
Wendi               1        4.35
Kari                1        4.35
Seth                1        4.35
Warren              1        4.35
Bernard             1        4.35
Shelby              1        4.35
Devon               1        4.35
Tonia               1        4.35
Janet               1        4.35

--------------------------------------------------
Para la columna MIDDLEINITIAL, los valores únicos son:
['T' 'W' 'Y' 'O' 'L' 'E' 'P' 'X' 'M' 'G' 'D' 'U' 'C' 'K']


Frecuencia  Porcentaje
MiddleInitial                        
E                       3       13.04
P                       3       13.04
D                       3       13.04
L                       2        8.70
O                       2        8.70
Y                       2        8.70
T                       1        4.35
W                       1        4.35
M                       1        4.35
X                       1        4.35
G                       1        4.35
U                       1        4.35
C                       1        4.35
K                       1        4.35

--------------------------------------------------
Para la columna LASTNAME, los valores únicos son:
['Fuller' 'Palmer' 'Cline' 'Nielsen' 'Stuart' 'Collins' 'Cook' 'Dyer'
 'King' 'Vang' 'Dickson' 'Chen' 'Marks' 'Buckley' 'Finley' 'Walton'
 'Franco' 'Bartlett' 'Moody' 'Riddle' 'Brewer' 'Mc Millan' 'Flowers']


Frecuencia  Porcentaje
LastName                         
Fuller              1        4.35
Palmer              1        4.35
Cline               1        4.35
Nielsen             1        4.35
Stuart              1        4.35
Collins             1        4.35
Cook                1        4.35
Dyer                1        4.35
King                1        4.35
Vang                1        4.35
Dickson             1        4.35
Chen                1        4.35
Marks               1        4.35
Buckley             1        4.35
Finley              1        4.35
Walton              1        4.35
Franco              1        4.35
Bartlett            1        4.35
Moody               1        4.35
Riddle              1        4.35
Brewer              1        4.35
Mc Millan           1        4.35
Flowers             1        4.35

--------------------------------------------------
Para la columna BIRTHDATE, los valores únicos son:
['1981-03-07 00:00:00.000' '1968-01-25 00:00:00.000'
 '1963-02-09 00:00:00.000' '1989-02-06 00:00:00.000'
 '1963-05-03 00:00:00.000' '1987-01-13 00:00:00.000'
 '1970-05-02 00:00:00.000' '1956-12-13 00:00:00.000'
 '1956-05-02 00:00:00.000' '1963-12-30 00:00:00.000'
 '1976-01-14 00:00:00.000' '1951-09-03 00:00:00.000'
 '1963-04-18 00:00:00.000' '1961-09-05 00:00:00.000'
 '1972-02-17 00:00:00.000' '1951-07-07 00:00:00.000'
 '1989-08-29 00:00:00.000' '1964-03-09 00:00:00.000'
 '1970-03-22 00:00:00.000' '1964-06-13 00:00:00.000'
 '1966-04-25 00:00:00.000' '1952-03-02 00:00:00.000'
 '1979-03-01 00:00:00.000']


Frecuencia  Porcentaje
BirthDate                                      
1981-03-07 00:00:00.000           1        4.35
1968-01-25 00:00:00.000           1        4.35
1963-02-09 00:00:00.000           1        4.35
1989-02-06 00:00:00.000           1        4.35
1963-05-03 00:00:00.000           1        4.35
1987-01-13 00:00:00.000           1        4.35
1970-05-02 00:00:00.000           1        4.35
1956-12-13 00:00:00.000           1        4.35
1956-05-02 00:00:00.000           1        4.35
1963-12-30 00:00:00.000           1        4.35
1976-01-14 00:00:00.000           1        4.35
1951-09-03 00:00:00.000           1        4.35
1963-04-18 00:00:00.000           1        4.35
1961-09-05 00:00:00.000           1        4.35
1972-02-17 00:00:00.000           1        4.35
1951-07-07 00:00:00.000           1        4.35
1989-08-29 00:00:00.000           1        4.35
1964-03-09 00:00:00.000           1        4.35
1970-03-22 00:00:00.000           1        4.35
1964-06-13 00:00:00.000           1        4.35
1966-04-25 00:00:00.000           1        4.35
1952-03-02 00:00:00.000           1        4.35
1979-03-01 00:00:00.000           1        4.35

--------------------------------------------------
Para la columna GENDER, los valores únicos son:
['F' 'M']


Frecuencia  Porcentaje
Gender                        
M               15       65.22
F                8       34.78

--------------------------------------------------
Para la columna HIREDATE, los valores únicos son:
['2011-06-20 07:15:36.920' '2011-04-27 04:07:56.930'
 '2012-03-30 18:55:23.270' '2014-03-06 06:55:02.780'
 '2014-11-16 22:59:54.720' '2013-06-22 13:20:18.080'
 '2016-07-10 06:22:00.670' '2014-10-14 23:12:53.420'
 '2013-04-17 14:48:02.700' '2012-07-23 15:02:12.640'
 '2016-08-10 15:59:30.360' '2011-11-03 00:44:25.390'
 '2011-12-12 10:43:52.940' '2016-07-11 00:57:58.340'
 '2014-07-13 06:50:11.790' '2017-02-10 11:21:26.650'
 '2010-07-11 09:23:03.490' '2010-02-22 21:55:48.620'
 '2013-11-12 19:25:04.390' '2013-03-24 08:36:50.020'
 '2016-01-24 02:13:24.340' '2015-11-25 18:18:23.480'
 '2010-05-12 02:52:30.370']


Frecuencia  Porcentaje
HireDate                                       
2011-06-20 07:15:36.920           1        4.35
2011-04-27 04:07:56.930           1        4.35
2012-03-30 18:55:23.270           1        4.35
2014-03-06 06:55:02.780           1        4.35
2014-11-16 22:59:54.720           1        4.35
2013-06-22 13:20:18.080           1        4.35
2016-07-10 06:22:00.670           1        4.35
2014-10-14 23:12:53.420           1        4.35
2013-04-17 14:48:02.700           1        4.35
2012-07-23 15:02:12.640           1        4.35
2016-08-10 15:59:30.360           1        4.35
2011-11-03 00:44:25.390           1        4.35
2011-12-12 10:43:52.940           1        4.35
2016-07-11 00:57:58.340           1        4.35
2014-07-13 06:50:11.790           1        4.35
2017-02-10 11:21:26.650           1        4.35
2010-07-11 09:23:03.490           1        4.35
2010-02-22 21:55:48.620           1        4.35
2013-11-12 19:25:04.390           1        4.35
2013-03-24 08:36:50.020           1        4.35
2016-01-24 02:13:24.340           1        4.35
2015-11-25 18:18:23.480           1        4.35
2010-05-12 02:52:30.370           1        4.35

--------------------------------------------------


Análisis inicial del archivo "employees":

1. Tipo de datos a cambiar: 

    - **BirthDate**: object -> datetime
    - **HireDate**: object -> datetime

2. Categorías a cambiar: 

    - **BirthDate**: dejar sólo la fecha
    - **Gender**: cambiar 'M' por Male y 'F' por Female
    - **HireDate**: dejar sólo la fecha
    - Cambiar nombre de las columnas:

        - **EmployeeID** -> Employee_Id
        - **BirthDate** -> Birth_Date
        - **CityID** -> City_Id
        - **HireDate** -> Hire_Date

    - Crear columna **Age** (edad del empleado) a partir de Birth_Date y **Seniority** (antigüedad del empleado en la empresa) a partir de Hire_Date

3. Columnas con valores nulos: no hay

4. Columnas a eliminar: 
    - Eliminar 'MiddleInitial' ya que no es necesaria para nuestro análisis
    - Concatenar las columnas 'FirstName' y 'LastName' en una sola, 'Full_Name' y eliminar las otras 2

In [7]:
ruta = ru.ruta_raw_data+"/products.csv" # Ruta del archivo CSV. Cambiar según sea necesario.
df_products = pd.read_csv(ruta)
sa.preanalisis(df_products)

El dataframe tiene 452 filas y 9 columnas
--------------------------------------------------
Tenemos un total de 0 valores duplicados
--------------------------------------------------
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 452 entries, 0 to 451
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   ProductID     452 non-null    int64  
 1   ProductName   452 non-null    object 
 2   Price         452 non-null    float64
 3   CategoryID    452 non-null    int64  
 4   Class         452 non-null    object 
 5   ModifyDate    452 non-null    object 
 6   Resistant     452 non-null    object 
 7   IsAllergic    452 non-null    object 
 8   VitalityDays  452 non-null    float64
dtypes: float64(2), int64(2), object(5)
memory usage: 31.9+ KB


None

--------------------------------------------------
No hay valores nulos
--------------------------------------------------
Los estadísticos descriptivos de las variables numéricas son:


count        mean         std     min        25%       50%  \
ProductID     452.0  226.500000  130.625419  1.0000  113.75000  226.5000   
Price         452.0   50.801471   28.616724  0.0449   26.50435   52.4995   
CategoryID    452.0    5.862832    3.271694  1.0000    3.00000    6.0000   
VitalityDays  452.0   26.030973   39.061200  0.0000    0.00000    0.0000   

                    75%       max  
ProductID     339.25000  452.0000  
Price          75.49645   99.8755  
CategoryID      9.00000   11.0000  
VitalityDays   52.50000  120.0000

--------------------------------------------------
Los estadísticos descriptivos de las variables categóricas son:


count unique                      top freq
ProductName   452    452  Napkin White - Starched    1
Class         452      3                   Medium  156
ModifyDate    452    452  2017-01-06 14:39:42.890    1
Resistant     452      3                  Durable  164
IsAllergic    452      3                    False  165

--------------------------------------------------
La distribución de frecuencias de las variables categóricas es:

Para la columna PRODUCTNAME, los valores únicos son:
['Flour - Whole Wheat' 'Cookie Chocolate Chip With' 'Onions - Cippolini'
 'Sauce - Gravy, Au Jus, Mix' 'Artichokes - Jerusalem'
 'Wine - Magnotta - Cab Sauv' 'Table Cloth - 53x69 Colour'
 'Halibut - Steaks' 'Rabbit - Whole' 'Scampi Tail' 'Garbage Bags - Clear'
 'Ezy Change Mophandle' 'Water, Tap' 'Beef - Top Sirloin'
 'Spoon - Soup, Plastic' 'Kellogs Special K Cereal'
 'Wine - White, Mosel Gold' 'Lamb - Whole, Fresh' 'Tea - Earl Grey'
 'Chocolate - Feathers' 'Kiwi' 'Rice - Jasmine Sented'
 'Crab - Imitation Flakes' 'Beer - Alexander Kieths, Pale Ale'
 'Soupcontfoam16oz 116con' 'Squid U5 - Thailand'
 'Chocolate - Compound Coating' 'Sobe - Tropical Energy' 'Spinach - Baby'
 'Cheese - Brie, Triple Creme' 'Yeast Dry - Fermipan' 'Lettuce - Treviso'
 'Beans - Wax' 'Mustard Prepared' 'Water - Mineral, Natural'
 'Table Cloth 54

Frecuencia  Porcentaje
ProductName                                       
Napkin White - Starched              1        0.22
Flour - Whole Wheat                  1        0.22
Cookie Chocolate Chip With           1        0.22
Onions - Cippolini                   1        0.22
Table Cloth 81x81 White              1        0.22
...                                ...         ...
Rabbit - Whole                       1        0.22
Halibut - Steaks                     1        0.22
Table Cloth - 53x69 Colour           1        0.22
Wine - Magnotta - Cab Sauv           1        0.22
Artichokes - Jerusalem               1        0.22

[452 rows x 2 columns]

--------------------------------------------------
Para la columna CLASS, los valores únicos son:
['Medium' 'Low' 'High']


Frecuencia  Porcentaje
Class                         
Medium         156       34.51
Low            153       33.85
High           143       31.64

--------------------------------------------------
Para la columna MODIFYDATE, los valores únicos son:
['2018-02-16 08:21:49.190' '2017-02-12 11:39:10.970'
 '2018-03-15 08:11:51.560' '2017-07-16 00:46:28.880'
 '2017-08-16 14:13:35.430' '2017-05-25 15:08:39.690'
 '2017-02-24 15:14:30.050' '2018-03-24 05:21:21.890'
 '2017-06-17 12:12:04.670' '2017-07-30 10:11:45.990'
 '2017-01-05 21:02:21.330' '2017-11-01 09:49:59.030'
 '2017-01-16 19:49:10.930' '2017-02-26 11:40:47.850'
 '2017-03-03 09:47:09.310' '2017-01-23 09:25:47.110'
 '2017-03-01 14:03:04.920' '2017-12-19 16:45:04.730'
 '2017-06-14 22:39:53.340' '2017-07-21 11:13:57.550'
 '2018-05-09 08:59:46.150' '2017-05-09 14:08:23.630'
 '2017-08-19 07:18:29.890' '2017-05-23 03:36:02.420'
 '2017-04-09 15:30:41.250' '2018-03-26 19:31:26.970'
 '2017-10-29 17:27:27.280' '2017-10-25 04:23:48.230'
 '2017-08-06 17:21:24.000' '2018-03-21 19:28:46.740'
 '2017-04-17 23:33:06.110' '2017-01-09 18:29:05.790'
 '2017-02-01 15:20:22.860' '2017-09-21 16:05:20.8

Frecuencia  Porcentaje
ModifyDate                                     
2017-01-06 14:39:42.890           1        0.22
2018-02-16 08:21:49.190           1        0.22
2017-02-12 11:39:10.970           1        0.22
2018-03-15 08:11:51.560           1        0.22
2017-07-13 00:46:43.380           1        0.22
...                             ...         ...
2017-06-17 12:12:04.670           1        0.22
2018-03-24 05:21:21.890           1        0.22
2017-02-24 15:14:30.050           1        0.22
2017-05-25 15:08:39.690           1        0.22
2017-08-16 14:13:35.430           1        0.22

[452 rows x 2 columns]

--------------------------------------------------
Para la columna RESISTANT, los valores únicos son:
['Durable' 'Unknown' 'Weak']


Frecuencia  Porcentaje
Resistant                        
Durable           164       36.28
Weak              148       32.74
Unknown           140       30.97

--------------------------------------------------
Para la columna ISALLERGIC, los valores únicos son:
['Unknown' 'False' 'True']


Frecuencia  Porcentaje
IsAllergic                        
False              165       36.50
True               157       34.73
Unknown            130       28.76

--------------------------------------------------


Análisis inicial del archivo "products":

1. Tipo de datos a cambiar: 

    - **ModifyDate**: object -> datetime
    - **VitalityDays**: float -> int

2. Categorías a cambiar: 

    - Cambiar nombre de las columnas:

        - **ProductID** -> Product_Id
        - **ProductName** -> Product_Name
        - **CategoryID** -> Category_Id
        - **ModifyDate** -> Modify_Date
        - **IsAllergic** -> Is_Allergic
        - **VitalityDays** -> Vitality_Days

    - **ModifyDate**: dejar sólo la fecha
    - **Is_Allergic**: cambiar True por Allergic, False por Not Allergic

3. Columnas con valores nulos: no hay

4. Columnas a eliminar: no son de interés para nuestro análisis

In [8]:
ruta = ru.ruta_raw_data+"/sales.csv" # Ruta del archivo CSV. Cambiar según sea necesario.
df_sales = pd.read_csv(ruta)
sa.preanalisis(df_sales)

El dataframe tiene 6758125 filas y 9 columnas
--------------------------------------------------
Tenemos un total de 0 valores duplicados
--------------------------------------------------
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6758125 entries, 0 to 6758124
Data columns (total 9 columns):
 #   Column             Dtype  
---  ------             -----  
 0   SalesID            int64  
 1   SalesPersonID      int64  
 2   CustomerID         int64  
 3   ProductID          int64  
 4   Quantity           int64  
 5   Discount           float64
 6   TotalPrice         float64
 7   SalesDate          object 
 8   TransactionNumber  object 
dtypes: float64(2), int64(5), object(2)
memory usage: 464.0+ MB


None

--------------------------------------------------
Los nulos por columna se distribuyen de la siguiente forma:


Valores_nulos  Porcentaje_nulos
SalesDate          67526               1.0

--------------------------------------------------
Los estadísticos descriptivos de las variables numéricas son:


count          mean           std  min        25%  \
SalesID        6758125.0  3.379063e+06  1.950903e+06  1.0  1689532.0   
SalesPersonID  6758125.0  1.199972e+01  6.632689e+00  1.0        6.0   
CustomerID     6758125.0  4.939567e+04  2.850504e+04  1.0    24707.0   
ProductID      6758125.0  2.265474e+02  1.304744e+02  1.0      114.0   
Quantity       6758125.0  1.300401e+01  7.209701e+00  1.0        7.0   
Discount       6758125.0  2.996787e-02  6.398096e-02  0.0        0.0   
TotalPrice     6758125.0  0.000000e+00  0.000000e+00  0.0        0.0   

                     50%        75%        max  
SalesID        3379063.0  5068594.0  6758125.0  
SalesPersonID       12.0       18.0       23.0  
CustomerID       49414.0    74075.0    98759.0  
ProductID          227.0      340.0      452.0  
Quantity            13.0       19.0       25.0  
Discount             0.0        0.0        0.2  
TotalPrice           0.0        0.0        0.0

--------------------------------------------------
Los estadísticos descriptivos de las variables categóricas son:


count   unique                      top freq
SalesDate          6690599  6670068  2018-04-08 09:42:48.360    3
TransactionNumber  6758125  6758125     RRBD7S603UIC73B631OF    1

--------------------------------------------------
La distribución de frecuencias de las variables categóricas es:

Para la columna SALESDATE, los valores únicos son:
['2018-02-05 07:38:25.430' '2018-02-02 16:03:31.150'
 '2018-05-03 19:31:56.880' ... '2018-01-18 12:05:16.950'
 '2018-02-26 08:13:37.850' '2018-04-17 15:26:30.270']


Frecuencia  Porcentaje
SalesDate                                      
2018-04-08 09:42:48.360           3         0.0
2018-03-02 20:58:43.080           3         0.0
2018-01-04 06:56:58.420           3         0.0
2018-05-09 08:54:35.030           3         0.0
2018-05-07 14:54:29.810           3         0.0
...                             ...         ...
2018-01-27 05:59:23.530           1         0.0
2018-05-06 04:45:13.030           1         0.0
2018-01-29 03:36:49.070           1         0.0
2018-01-16 12:25:15.610           1         0.0
2018-03-08 21:02:34.480           1         0.0

[6670068 rows x 2 columns]

--------------------------------------------------
Para la columna TRANSACTIONNUMBER, los valores únicos son:
['FQL4S94E4ME1EZFTG42G' '12UGLX40DJ1A5DTFBHB8' '5DT8RCPL87KI5EORO7B0' ...
 '7VZJ77UGBSX2SNCC1GFW' '3JTF5IF0YJLLXF9IRS8X' 'RRBD7S603UIC73B631OF']


Frecuencia  Porcentaje
TransactionNumber                           
RRBD7S603UIC73B631OF           1         0.0
FQL4S94E4ME1EZFTG42G           1         0.0
12UGLX40DJ1A5DTFBHB8           1         0.0
5DT8RCPL87KI5EORO7B0           1         0.0
R3DR9MLD5NR76VO17ULE           1         0.0
...                          ...         ...
S09LSXPASD6KXLHZRZWL           1         0.0
P73F1OA46H2JE3EF83ZM           1         0.0
AO360LVKUYJSSGL5T5MF           1         0.0
E2JF4LIQNYTDKDBGMJS0           1         0.0
VOP9A7Y4C5XSM2LLT0UJ           1         0.0

[6758125 rows x 2 columns]

--------------------------------------------------


Análisis inicial del archivo "sales":

1. Tipo de datos a cambiar: 

    - **SalesDate**: object -> datetime

2. Categorías a cambiar: 

    - Cambiar nombre de las columnas:

        - **SalesID** -> Sales_Id
        - **SalesPersonID** -> Sales_Person_Id
        - **CustomerID** -> Customer_Id
        - **ProductID** -> Product_Id
        - **SalesDate** -> Sales_Date

    - **Sales_Date**: dejar sólo la fecha y crear una nueva columna sólo con la hora (no los minutos) para ver las ventas por horas

    - Crear nueva columna Total_Price obteniendo el precio del producto del archivo 'products' y aplicando el descuento.

3. Columnas con valores nulos: 

    - **SalesDate**. 67526 valores nulos

4. Columnas a eliminar: 

    - **TransactionNumber**: no aporta valor a nuestro análisis
    - **TotalPrice**: todos los valores son ceros